watertap/property_models/unit_specific/activated_sludge/asm3_reactions.py

```diff
@@ -0,0 +1,863 @@
+                elif r == "R2":
+                    # R2: Aerobic storage of S_S
+                    return b.reaction_rate[r] == pyo.units.convert(
+                        b.params.k_STO["20C"]
+                        * (
+                            b.conc_mass_comp_ref["S_O"]
+                            / (b.params.K_O2 + b.conc_mass_comp_ref["S_O"])
+                        )
+                        * (
+                            b.conc_mass_comp_ref["S_S"]
+                            / (b.params.K_S + b.conc_mass_comp_ref["S_S"])
+                        )
+                        * b.conc_mass_comp_ref["X_H"],
+                        to_units=pyo.units.kg / pyo.units.m**3 / pyo.units.s,
+                    )
+                elif r == "R3":
+                    # R3: Anoxic storage of S_S
+                    return b.reaction_rate[r] == pyo.units.convert(
+                        b.params.k_STO["20C"]
+                        * b.params.eta_NOX
+                        * (
+                            b.params.K_O2
+                            / (b.params.K_O2 + b.conc_mass_comp_ref["S_O"])
+                        )
+                        * (
+                            b.conc_mass_comp_ref["S_NOX"]
+                            / (b.params.K_NOX + b.conc_mass_comp_ref["S_NOX"])
+                        )
+                        * (
+                            b.conc_mass_comp_ref["S_S"]
+                            / (b.params.K_S + b.conc_mass_comp_ref["S_S"])
+                        )
+                        * b.conc_mass_comp_ref["X_H"],
+                        to_units=pyo.units.kg / pyo.units.m**3 / pyo.units.s,
+                    )
+                elif r == "R4":
+                    # R4: Aerobic growth
+                    return b.reaction_rate[r] == pyo.units.convert(
+                        b.params.mu_H["20C"]
+                        * (
+                            b.conc_mass_comp_ref["S_O"]
+                            / (b.params.K_O2 + b.conc_mass_comp_ref["S_O"])
+                        )
+                        * (
+                            b.conc_mass_comp_ref["S_NH4"]
+                            / (b.params.K_NH4 + b.conc_mass_comp_ref["S_NH4"])
+                        )
+                        * (
+                            b.state_ref.alkalinity
+                            / (b.params.K_ALK + b.state_ref.alkalinity)
+                        )
+                        * (b.conc_mass_comp_ref["X_STO"] / b.conc_mass_comp_ref["X_H"])
+                        / (
+                            b.params.K_STO
+                            + b.conc_mass_comp_ref["X_STO"]
+                            / b.conc_mass_comp_ref["X_H"]
+                        )
+                        * b.conc_mass_comp_ref["X_H"],
+                        to_units=pyo.units.kg / pyo.units.m**3 / pyo.units.s,
+                    )
+                elif r == "R5":
```

Comment:
Double check this equation - I think you may be missing some NOX terms

Reply:
Thank you! Updated the NOX term